doc/api/async_hooks.md

```diff
@@ -2,7 +2,10 @@
 
 <!--introduced_in=v8.1.0-->
 
-> Stability: 1 - Experimental
+> Stability: 1 - Experimental. We plan to remove this API
```

Comment:
Really? I'm not aware of any plan for removal.

Reply:
That’s what this PR is meant to declare. We’re agreeing that we want to eventually remove it, and the specifics of how will be determined.

Reply:
Ok, in that cases I don't agree.
I see no benefit in removing something from public API surface which is need internally without providing a replacement. This only creates unneeded friction.
As I see no path to get a replacement as of now I see a similar topic here as in domains.

Reply:
> I see no benefit in removing something from public API surface which is need internally without providing a replacement.

`async_hooks` _exposes_ internals. That’s why it shouldn’t remain a public API; but also why it’s fine to continue existing as an internal API.

The replacements are `AsyncLocalStorage` and Diagnostics Channel. If there are use cases currently served by `async_hooks` that are unachievable with those APIs, then those use cases are ones we should try to address before `async_hooks` is removed, either via `AsyncLocalStorage` and Diagnostics Channel or some other APIs. The point of this PR is to start that process: to encourage users depending on `async_hooks` to start migrating off of it, and report issues that they encounter so that collaborators can address them, by adding new capabilities to `AsyncLocalStorage` or Diagnostics Channel or other APIs that we consider stable or on a path toward stability.